### PR TITLE
javascript/armstrong-numbers: initial mentor notes

### PR DESCRIPTION
**Pull request by SleeplessByte**

### CODE REVIEW

**Comment by SleeplessByte** [comment]
Build failing has nothing to do with this PR.

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+1. convert the input `number` to a list of digits
+2. `reduce` the list of digits to a _sum_ of each digit to the power of the number of digitis
+3. compare the input to the summation
+
+```javascript
+const digits = [...input.toString(10)].map(Number)
+const n = digits.length
+return input === digits.reduce((result, digit) => result + digit ** n, 0)
+```
```
Comment (tejasbubane) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]: Can we put this in a function? It is better to have a `complete` solution IMO.

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+It's cleaner to extract out the length _before_ iteration, but its also fine
+to rely on the engine to optimise it. The length can either be extracted from
+the `digits` array, or from `self`, the fourth argument of the `.reduce(...)`
+callback.
+
+### Common suggestions
+
+- Its more idiomatic to explicitly convert to a string, instead of using `+ ''`
+- Instead of using `.map(n => +n)`, the `Number` function can be passed in,
+  without an anonymous function.
```
Comment (tejasbubane) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]: like - `.map(Number)`

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md (hunk 1 above)
Comment (SleeplessByte) [reply tracks/javascript/exercises/armstrong-numbers/mentoring.md]: Woops. I don't know why it's not ❌

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md (hunk 2 above)
Comment (SleeplessByte) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]:
```suggestion
  without an anonymous function: `.map(Number)`.
```

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+### Reasonable solution
+
+The solution should consist of the following three steps:
+
+1. convert the input `number` to a list of digits
+2. `reduce` the list of digits to a _sum_ of each digit to the power of the number of digitis
+3. compare the input to the summation
+
+```javascript
+const digits = [...input.toString(10)].map(Number)
```
Comment (SleeplessByte) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]:
```suggestion
function steps(input) {
  const digits = [...input.toString(10)].map(Number)
```

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const n = digits.length
```
Comment (SleeplessByte) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]:
```suggestion
  const n = digits.length
```

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+return input === digits.reduce((result, digit) => result + digit ** n, 0)
```
Comment (SleeplessByte) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]:
```suggestion
  return input === digits.reduce((result, digit) => result + digit ** n, 0)
}
```

**Comment by SleeplessByte** [comment]
@ihid can you restart this one? (CI)

**Comment by tejasbubane** [comment]
@SleeplessByte I restarted & it failed again. You'll have to rebase since iHiD fixed it in https://github.com/exercism/website-copy/commit/81f3240d772016597737c2c8c32a8b5be61ce93d on Dec 20 (after your commit).

File: tracks/javascript/exercises/armstrong-numbers/mentoring.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+### Reasonable solution
+
+The solution should consist of the following three steps:
+
+1. convert the input `number` to a list of digits
+2. `reduce` the list of digits to a _sum_ of each digit to the power of the number of digitis
```
Comment (tejasbubane) [inline tracks/javascript/exercises/armstrong-numbers/mentoring.md]:
```suggestion
2. `reduce` the list of digits to a _sum_ of each digit to the power of the number of digits
```

**Pull request merged**